### PR TITLE
Engine Impruvments

### PR DESCRIPTION
**Pull request by EndrII**
Need to solve #49 before merge this PR.

### CODE REVIEW

**Review by IgorekLoschinin — APPROVED** [review]

File: src/Core/SnakeProject/iworld.cpp [hunk 1]
```diff
@@ -109,9 +163,31 @@ bool IWorld::removeItem(int id) {
     return true;
 }
 
-bool IWorld::removeAnyItemFromGroup(const QString &group) {
-    auto anyObject = _itemsGroup.value(group);
-    return removeItem(anyObject);
+int IWorld::removeAnyItemFromGroup(const QString &group) {
+    int anyObjectId = _itemsGroup.value(group);
+    if (!removeItem(anyObjectId)) {
+        return 0;
```
[the remark this answers is not in the document]
Comment (EndrII) [reply src/Core/SnakeProject/iworld.cpp:169]: This method should be return removed item id. If the item not removed then return 0.

File: src/Core/SnakeProject/iworld.cpp [hunk 2]
```diff
@@ -132,6 +216,7 @@ const QString &IWorld::hdrMap() const {
 
 void IWorld::worldChanged(const WorldObjects &objects) {
 
+    Diff diff;
     for (auto it = objects.begin(); it != objects.end(); ++it) {
```
[the remark this answers is not in the document]
Comment (EndrII) [reply src/Core/SnakeProject/iworld.cpp:220]: No, a cbegin method return const iterators only but a begin method return general iterator for general objects and const iterator for const objects. In This situation  the 'objects' object already constant and  its begin method return constant iterator.

The cbegin method used in non constant objects for getting constant iterators.

File: src/Core/private/engine.h [hunk 3]
```diff
@@ -70,6 +72,26 @@ class Engine : public QObject {
...
+
+    /**
+     * @brief setMenu This method sets new menu object.
+     * @param newMenu
+     */
+    void setMenu(QObject *newMenu);
+
+    /**
+     * @brief prepareLvlProgress This method return rurrent progress of the loading lvl.
+     * @return
```
Comment (EndrII) [inline src/Core/private/engine.h:90]:
```suggestion
     * @return current progress of loading new level on the world. progress range is 0 - 100 
```

[the remark this answers is not in the document]
Comment (EndrII) [reply src/Core/private/engine.h]: done

File: src/Core/private/pluginloader.cpp [hunk 4]
```diff
@@ -2,33 +2,34 @@
...
+IWorld* PluginLoader::load(const QString &pluginPath) {
     QLibrary lib(pluginPath);
 
     if (!lib.load()) {
 
         QuasarAppUtils::Params::log("Fail to load game module. Message: " + lib.errorString(),
                                     QuasarAppUtils::Error);
 
-        return nullptr;
+        return {};
```
Comment (EndrII) [inline src/Core/private/pluginloader.cpp:20]:
```suggestion
        return nullptr;
```

File: src/Core/private/pluginloader.cpp [hunk 5]
```diff
@@ -2,33 +2,34 @@
...
 
-    if (!func) {
+    if (!worldFunc) {
         QuasarAppUtils::Params::log("Fail to load game module."
                                     " Message: Failed to find a instance function in the %0 module",
                                     QuasarAppUtils::Error);
 
         lib.unload();
-        return nullptr;
+        return {};
```
Comment (EndrII) [inline src/Core/private/pluginloader.cpp:31]:
```suggestion
        return nullptr;
```

**Review by IgorekLoschinin — APPROVED** [review]

**Pull request merged**